pkg/splunk/enterprise/configuration.go

```diff
@@ -639,3 +642,69 @@ func updateSplunkPodTemplateWithConfig(podTemplateSpec *corev1.PodTemplateSpec,
 		podTemplateSpec.Spec.Containers[idx].Env = env
 	}
 }
+
+// LogSmartStoreVolumes logs smartstore volumes
+func LogSmartStoreVolumes(volumeList []enterprisev1.VolumeSpec) {
+	scopedLog := logC.WithName("LogtSmartStoreVolumes")
+	//var temp string
+	for _, volume := range volumeList {
+		scopedLog.Info("Volume: ", "name: ", volume.Name, "endpoint: ", volume.S3Endpoint)
+	}
+}
+
+// LogSmartStoreIndexes logs smartstore indexes
+func LogSmartStoreIndexes(indexList []enterprisev1.IndexSpec) {
+	scopedLog := logC.WithName("LogSmartStoreIndexes")
+	for _, index := range indexList {
+		scopedLog.Info("Index: ", "name: ", index.Name, "location: ", index.RemoteLocation)
+	}
+}
+
+// isSmartstoreEnabled checks and returns true if smartstore is configured
+func isSmartstoreConfigured(smartstore *enterprisev1.SmartStoreSpec) bool {
+	if smartstore == nil {
+		return false
+	}
+
+	return !(smartstore.IndexList == nil && smartstore.VolList == nil)
+}
+
+// ValidateSplunkSmartstoreSpec checks and validates the smartstore config
+func ValidateSplunkSmartstoreSpec(smartstore *enterprisev1.SmartStoreSpec) error {
+	// Smartstore is an optional config (at least) for now
+	if !isSmartstoreConfigured(smartstore) {
+		return nil
```

Comment:
Its better to log an error msg saying something like  - "Smartstore is not configured" or something along those lines.

Reply:
This is not an error, as not configuring the smart store is perfectly fine(at least for this release)

Reply:
Ok, makes sense